### PR TITLE
[CALCITE-5590] NullPointerException when converting 'in' expression that is used inside select list and group by

### PR DESCRIPTION
**Pull request by dssysolyatin**
Note that almost all deleted lines are from my incorrect previous fix for task [CALCITE-5209](https://issues.apache.org/jira/browse/CALCITE-5209)

### CODE REVIEW

**Review by mihaibudiu — APPROVED** [review]
This looks a bit like magic, but it seems to work based on the tests.

**Review by caicancai — APPROVED** [review]
LGTM

**Comment by dssysolyatin** [comment]
@mihaibudiu @caicancai Thanks for reviewing. I made one more small change—I added a test case from the task https://issues.apache.org/jira/browse/CALCITE-4549 to ensure that it also works.

File: core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java [hunk 1]
```diff
@@ -523,6 +523,16 @@ public static void checkActualAndReferenceFiles() {
     sql(sql).ok();
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-4549">[CALCITE-4549]
+   * IndexOutOfBoundsException when group view by a sub query</a>. */
+  @Test void testGroupView() {
```
Comment (caicancai) [inline core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java:529]: Maybe the test name could be improved, but it doesn't seem to matter.

**Pull request merged**